### PR TITLE
fix bug where an edge case exception causes a query to not be removed…

### PR DESCRIPTION
**Pull request by eriktaubeneck**
… from active running queries

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved reliability of query execution by ensuring proper cleanup of running queries, even in the event of an error.
	- Added tests to verify the behavior of the query manager when exceptions occur during query execution.

### CODE REVIEW

File: sidecar/app/query/base.py [hunk 1]
```diff
@@ -201,8 +201,11 @@ def run_query(self, query: Query):
             )
 
         self.running_queries[query.query_id] = query
-        query.start()
-        del self.running_queries[query.query_id]
+        try:
+            query.start()
+        finally:
+            # always remove this
+            del self.running_queries[query.query_id]
```
Comment (akoshelev) [inline sidecar/app/query/base.py:208]: is there a way to test this behavior to make sure we don't miss this edge case by accidentally removing this code?

File: sidecar/tests/app/query/test_base.py [hunk 2]
```diff
@@ -119,3 +119,22 @@ def fake_start():
     ) as mock_start:
         query_manager.run_query(query)
         mock_start.assert_called_once()
+
+
+def test_query_manger_run_query_exception():
```
Comment (akoshelev) [inline sidecar/tests/app/query/test_base.py:124]:
```suggestion
def test_query_manager_run_query_exception():
```

**Review by akoshelev — APPROVED** [review]
nice test!

File: sidecar/app/query/base.py (hunk 1 above)
Comment (eriktaubeneck) [reply sidecar/app/query/base.py:208]: added!
